src/handlers.rs

```diff
@@ -393,6 +397,147 @@ where
                 maybe_await!(wallet.verify_proof(&psbt, &msg, max_confirmation_height))?;
             Ok(json!({ "spendable": spendable }))
         }
+
+        // Payjoin Logic goes here
+        SendPayjoin { uri } => {
+            // convert the bip21 uri into a payjoin uri, and handle error if necessary
+            let uri = payjoin::Uri::try_from(uri).map_err(|e| {
+                Error::Generic(format!("Unable to convert BIP21 into Payjoin URI: {}", e))
+            })?;
+
+            // ensure uri is payjoin capable
+            let uri = uri
+                .check_pj_supported()
+                .map_err(|e| Error::Generic(format!("Payjoin not supported: {}", e)))?;
+
+            // ensure amount of satoshis is specified in uri, handle error if not
+            let sats = match uri.amount {
+                Some(amount) => Ok(amount.to_sat()),
+                None => Err(Error::Generic("URI was empty".to_string())),
+            }?;
+
+            // call wallet build tx to create psbt
+            let mut tx_builder = wallet.build_tx();
+
+            // assuming uri address is of type address
+            let scrm = uri.address.script_pubkey();
+
+            // create tx with payjoin uri
+            tx_builder
+                .add_recipient(scrm.clone(), sats)
+                .fee_rate(FeeRate::from_sat_per_vb(1.0))
+                .enable_rbf();
+
+            // create psbt
+            let (mut psbt, _tx_details) = tx_builder
+                .finish()
+                .map_err(|e| Error::Generic(format!("Unable to build transaction: {}", e)))?;
+
+            // check to make sure it is finalized (sign will return bool true )
+
+            // set signoptions
+            let mut sign_options = SignOptions::default();
+            sign_options.trust_witness_utxo = true;
+
+            // check + sign psbt
+            let finalized = wallet.sign(&mut psbt, sign_options)?;
```

Comment:
Default SignOptions seem to work now that we've got the right data to sign available

```suggestion
            let finalized = wallet.sign(&mut psbt, SignOptions::default())?;
```